zrml/court/src/tests.rs

```diff
@@ -3099,6 +3116,23 @@ fn handle_inflation_works() {
     });
 }
 
+#[test]
+fn handle_inflationis_noop_if_yearly_inflation_is_zero() {
+    ExtBuilder::default().build().execute_with(|| {
+        <YearlyInflation<Runtime>>::kill();
+
+        let inflation_period = InflationPeriod::get();
+        run_blocks(inflation_period);
+        let now = <frame_system::Pallet<Runtime>>::block_number();
+
+        // save current storage root
+        let tmp = storage_root(StateVersion::V1);
+        // handle_inflation is a noop for the storage
+        Court::handle_inflation(now);
+        assert_eq!(tmp, storage_root(StateVersion::V1));
```

Comment:
Two things:

- Why are you manually calling `handle_inflation` when it is already triggered by running the blocks?
- Storing `tmp` must happen before running the blocks. This is because you're starting on block `1`, so when you run `InflationPeriod::get()` many blocks, you end up overshooting by one block, which means you're checking the wrong block when you're manually rerunning `handle_inflation`.

I guess a different solution would be to not run `InflationPeriod::get()` many block but instead run until the next time inflation is distributed.

```suggestion
        let inflation_period = InflationPeriod::get();
        // save current storage root
        let tmp = storage_root(StateVersion::V1);
        run_blocks(inflation_period);
        let now = <frame_system::Pallet<Runtime>>::block_number();

        // handle_inflation is a noop for the storage
        Court::handle_inflation(now);
        assert_eq!(tmp, storage_root(StateVersion::V1));
```

Reply:
You are right. It was because I was not concentrated anymore yesterday. We don't need to run blocks in this test. I fixed it here 3e60e5fb3faea24ddbc601853c8b528f311df462